### ISSUE TITLE
Mailing addresses should not contain punctuation characters

### ISSUE THREAD

**Issue by prinzdezibel** [reactioncommerce/reaction#3721]
### Expected behavior
Mailing addresses should not contain punctuation characters according to this link: [https://www.wikihow.com/Write-an-Apartment-Address](https://www.wikihow.com/Write-an-Apartment-Address) , Part 3 

Assuming the above reference is valid, I'd expect not to see punctuation characters in any of the addresses.

### Actual behavior
Most addresses contain punctuation characters.

### Steps to reproduce the behavior
- As a user, place an order.
- See the city, state and zip-code is separated by commas in different places. See screenshots below  for examples:
![cart_checkout](https://user-images.githubusercontent.com/1733229/36063307-6bb8b512-0e7b-11e8-91cd-bafb000455ff.jpg)
![cart_completed](https://user-images.githubusercontent.com/1733229/36063308-6d9153c6-0e7b-11e8-82a9-0e71b02acc09.jpg)
![maildev](https://user-images.githubusercontent.com/1733229/36063310-70495b7c-0e7b-11e8-8f69-cf27f1b9dd5c.jpg)
![reaction](https://user-images.githubusercontent.com/1733229/36063312-720ddece-0e7b-11e8-9c11-a3433b84d789.jpg)

**Comment by zenweasel** [comment]
@prinzdezibel I'm not sure I would consider WIkiHow the canonical reference on **anything** but I do agree it should be consistent. However it's totally wrong that the apartment should never contain punctuation. As a life-long apartment dweller I can tell you that nobody puts "Apartment 5" and if you force people to do that you will lose customers. Customers know better than we do what to put to get stuff properly delivered to their address.

**Comment by zenweasel** [comment]
I guess what I am saying is that what is "right" and what we should force people to do are two separate things.

**Comment by prinzdezibel** [comment]
@zenweael: Agree, that this Wiki is not a reference on anything, of course. It served me to get a clue on how US addresses are supposed to be formatted in general, because I had no clue. 
> However it's totally wrong that the apartment should never contain punctuation.
I think this was not meant by the wiki. I think they wanted to state that punctuation should not be used to separate city, street and apartment from each other. As I've understood, it's still ok to use it as abbreviation, like in APT.

**Comment by mattt416** [comment]
I've used this link in the past, it's quite extensive:

https://www.canadapost.ca/tools/pg/manual/PGaddress-e.asp#1417197

**Comment by zenweasel** [comment]
I think we need to rewrite the title/details for this issue so that it's clear that we are make the way that addresses display consistent. It's not clear to me what's broken from the current title and description.

**Comment by prinzdezibel** [comment]
done.

**Comment by aldeed** [comment]
Note that the path forward on this issue will be to use the new `Address` component everywhere: https://designsystem.reactioncommerce.com/#!/Address
